src/tests/classes.test.js

```diff
@@ -1,3 +1,326 @@
+    expect(response.status).toBe(404);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Invalid ID should return status 400', async () => {
+    const response = await request(app).put('/api/classes/123456').send();
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('ID not found should return status 404', async () => {
+    const updatedClass = {
+      day: 'Monday', hour: '20:00',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510629').send(updatedClass);
+    expect(response.status).toBe(404);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Duplicated class should return status 400', async () => {
+    const duplicatedClass = {
+      day: 'Sunday', hour: '20:00',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510620').send(duplicatedClass);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Day not valid should return status 400', async () => {
+    const dayInvalid = {
+      day: 'Miernes',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510620').send(dayInvalid);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Hour not valid should return status 400', async () => {
+    const hourInvalid = {
+      hour: 'asd023',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510620').send(hourInvalid);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('If no changes were made, return status 400', async () => {
+    const notChanged = { day: 'Thursday' };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510624').send(notChanged);
+    expect(response.status).toBe(200);
+    expect(response.body.data).toBeDefined();
+    expect(response.body.error).toBeFalsy();
+  });
+
+  test('If the endpoint is wrong, return 400', async () => {
+    const response = await request(app).put('/api/classessss').send();
+    if (response.body.data === 0) {
+      expect(response.status).toBe(400);
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.error).toBeTruthy();
+    }
+  });
+
+  test('Updated class should return status 200', async () => {
+    const validClass = {
+      day: 'Monday',
+      hour: '10:00',
+      slots: 15,
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510621').send(validClass);
+    expect(response.status).toBe(201);
```

Comment:
The response is setted toBe 201 and the status is made for POST. Maybe it should be 200 for being PUT.

Reply:
i agreed with Mateo, it should be status 200 because 201 is when the class is created